astropy/vo/client/tests/test_conesearch.py

```diff
@@ -96,15 +99,12 @@ def test_sky_coord(self):
 
     def test_timeout(self):
         """Test time out error."""
-        try:
+        with pytest.raises(VOSError) as e:
             with data.conf.set_temp('remote_timeout', 0.001):
-                tab = conesearch.conesearch(
+                conesearch.conesearch(
                     SCS_CENTER, SCS_RADIUS, pedantic=self.pedantic,
                     verbose=self.verbose, catalog_db=self.url, cache=False)
-        except VOSError as e:
-            assert 'timed out' in str(e), 'test_timeout failed'
-        else:
-            raise Exception('test_timeout failed')
+            assert 'timed out' in e.info, 'test_timeout failed'
```

Comment:
You mean, why add "test_timeout failed"? That is the error message to go with `AssertionError` if it is raised. Or did I misunderstand your question?